srk/src/interpretation.ml

```diff
@@ -253,6 +256,7 @@ let select_implicant interp ?(env=Env.empty) phi =
         | `Neg -> mk_neg srk t_term
       in
       (term, t_impl)
+    | `Select _ -> assert false
```

Comment:
W.r.t. finding implicant involving/evaluating select: during our previous discussion I thought we were not happy with my definition of concrete arrays. You will notice that the type 'a value doesn't support arrays and thus we cannot currently handle interpretations with array variables. It follows that we can't evaluate select and thus we also can't find an implicant involving select.

I am happy to bring the previous definition back (with modifications). If I recall, there were two issues with the previous definition: (1) our language has the sort integer-arrays but I was interpreting them using arrays in which the elements are rational numbers. (2) I had a notion of extensional equality.

W.r.t. 1: I did this because we are also mapping integer sorted variables to rational numbers. I did this to follow that same convention but I am just as happy to use integers here. Let me know which you think is best.

W.r.t. 2: I will just remove any notion of equality for the concrete arrays.

Reply:
Actually, I'm ok with leaving it out.  The choice of not implementing functionality that you don't currently need is reasonable.
(The larger issue with the previous implementation is that it made an arbitrary choice that arrays are piecewise constant functions.  The implementation of function values in interpretations gives a more  general solution, in which function values are expressions with parameters -- so any [definable](https://en.wikipedia.org/wiki/Definable_set) function has a representation as a function value.)